app/src/pages/inside/uniqueErrorsPage/modals/uniqueErrorsAnalyzeModal.jsx

```diff
@@ -35,30 +46,72 @@ const cx = classNames.bind(styles);
   initialValues: { removeNumbers: false },
 })
 @injectIntl
+@connect(
+  (state) => ({
+    projectId: activeProjectSelector(state),
+  }),
+  {
+    showNotification,
+    showDefaultErrorNotification,
+    updateLaunchLocallyAction,
+  },
+)
 export class UniqueErrorsAnalyzeModal extends Component {
   static propTypes = {
     intl: PropTypes.object.isRequired,
     handleSubmit: PropTypes.func.isRequired,
-    data: PropTypes.object,
+    data: PropTypes.shape({
+      launch: PropTypes.object.isRequired,
+    }).isRequired,
+    projectId: PropTypes.string.isRequired,
+    showNotification: PropTypes.func,
+    showDefaultErrorNotification: PropTypes.func,
+    updateLaunchLocallyAction: PropTypes.func,
   };
 
   static defaultProps = {
     handleSubmit: () => {},
-    data: {},
+    showNotification: () => {},
+    showDefaultErrorNotification: () => {},
+    updateLaunchLocallyAction: () => {},
+  };
+
+  onSubmit = ({ removeNumbers }) => {
+    const { projectId, data } = this.props;
+    const launch = data.launch;
+    fetch(URLS.runUniqueErrorAnalysis(projectId), {
+      method: 'POST',
+      data: {
+        launchId: launch.id,
+        removeNumbers,
+      },
+    })
+      .then(({ message }) => {
+        this.props.showNotification({
+          message,
+          type: NOTIFICATION_TYPES.SUCCESS,
+        });
+        const analysing = launch.analysing;
+        const item = {
+          ...launch,
+          analysing: [...analysing, ANALYZER_TYPES.CLUSTER_ANALYSER],
+        };
+        this.props.updateLaunchLocallyAction(item);
```

Comment:
In case of running cluster analysing from unique errors page we don't have items on launches page.
So I think the different action should be used here in this case. This action can be replaced with a callback, it will be `this.props.updateLaunchLocallyAction` if we call this modal from launches page, or it can be smth else (may be parent launch update) if we use it from unique errors page. You can `dispatch` `fetchParentLaunchSuccessAction` with necessary payload.